addon/-private/array.ts

```diff
@@ -147,16 +73,94 @@ class TrackedArray<T = unknown> {
     thisArg?: unknown
   ): TrackedArray<T> | TrackedArray<U> {
     return mapfn
-      ? createArrayProxy(Array.from(iterable, mapfn, thisArg))
-      : createArrayProxy(Array.from(iterable));
+      ? new TrackedArray(Array.from(iterable, mapfn, thisArg))
+      : new TrackedArray(Array.from(iterable));
   }
 
   static of<T>(...arr: T[]): TrackedArray<T> {
-    return createArrayProxy(arr);
+    return new TrackedArray(arr);
   }
 
   constructor(arr: T[] = []) {
-    return createArrayProxy(arr.slice());
+    let clone = arr.slice();
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    let self = this;
+
+    let boundFns = new Map();
+
+    return new Proxy(clone, {
+      get(target, prop, receiver) {
+        let index = convertToInt(prop);
+
+        if (index !== null) {
+          self.storageFor(index);
```

Comment:
@NullVoxPopuli - shouldn't this be `self.readStorageFor(index)` (like in `tracked-object`)? And `readStorageFor` should also be changed to `getValue(storage)` instead of return `storage`. Or am I missing something?

Reply:
it's been a minute, but I think it's just a method naming inconsistency?

Reply:
Nope, check the similar method in `object.ts`. It does `getValue(storage)` (which in the `array` case it doesn't do).

Reply:
it does for the collection reference